### ISSUE TITLE
unnecessary type arguments infertypeargs

### ISSUE THREAD

**Issue by jiaopengzi** [golangci/golangci-lint#5488]
### Welcome

- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

go install

### Your feature request related to a problem? Please describe

Hello,

After upgrading to Go version `1.24.0`, we noticed that the generic type parameters no longer need to be explicitly specified as they can be inferred automatically.

Previously, the code would be written as follows:

```go
func f[T any](T) {}
	
func _() {
	f[string]("foo")
}
```

Now, it can be written as:

```go
func f[T any](T) {}
	
func _() {
	f("foo") // string could be inferred
}
```

Official reference: [link](https://cs.opensource.google/go/x/tools/+/refs/tags/gopls/v0.18.1:gopls/internal/analysis/infertypeargs/infertypeargs.go)

If the old style is still used, the editor already provides a hint: `unnecessary type arguments infertypeargs`.

However, running `golangci-lint run` currently does not detect this issue. We would appreciate it if this functionality could be added.

golangci-lint:`v1.64.5`

Thank you very much.

### Describe the solution you'd like

We hope that `golangci-lint run` will provide a warning for the issue `unnecessary type arguments infertypeargs`.

### Describe alternatives you've considered

I apologize for my lack of expertise in this area. I can only raise the issue and currently do not have the capability to solve it.

### Supporter

- [ ] I am a sponsor through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
> After upgrading to Go version 1.24.0, we noticed that the generic type parameters no longer need to be explicitly specified as they can be inferred automatically.

The type inference exists since the introduction of generics (go1.18), it is not related to go1.24 or new.

---

Note: there is a GitHub bug with converting an issue to discussion: https://github.com/orgs/community/discussions/152627

**Comment by ldez** [comment]
Ok, you are referring to a gopls analyzer.

https://github.com/golangci/golangci-lint/issues/686#issuecomment-2687812485
